src/dodal/devices/pressure_jump_cell.py

```diff
@@ -0,0 +1,236 @@
+    BUSY = "Busy"
+
+
+class TimerState(str, Enum):
+    TIMEOUT = "TIMEOUT"
+    COUNTDOWN = "COUNTDOWN"
+
+
+class StopState(str, Enum):
+    CONTINUE = "CONTINUE"
+    STOP = "STOP"
+
+
+class ValveControlRequest(str, Enum):
+    OPEN = "Open"
+    CLOSE = "Close"
+    RESET = "Reset"
+    ARM = "Arm"
+    DISARM = "Disarm"
+
+
+class PumpMotorControlRequest(str, Enum):
+    ENABLE = "Enable"
+    DISABLE = "Disable"
+    RESET = "Reset"
+    FORWARD = "Forward"
+    REVERSE = "Reverse"
+
+
+class PumpMotorDirectionState(str, Enum):
+    ZERO = "0"
+    FORWARD = "Forward"
+    REVERSE = "Reverse"
+    THREE = "3"
+    FOUR = "4"
+    FIVE = "5"
+    SIX = "6"
+    SEVEN = "7"
+
+
+class ValveState(str, Enum):
+    FAULT = "Fault"
+    OPEN = "Open"
+    OPENING = "Opening"
+    CLOSED = "Closed"
+    CLOSING = "Closing"
+
+
+class FastValveState(str, Enum):
+    FAULT = "Fault"
+    OPEN = "Open"
+    OPEN_ARMED = "Open Armed"
+    CLOSED = "Closed"
+    CLOSED_ARMED = "Closed Armed"
+    NONE5 = "Unused"
+
+
+class LimitSwitchState(str, Enum):
+    OFF = "Off"
+    ON = "On"
+
+
+class AllValvesControl(StandardReadable):
+    """
+    valves 2, 4, 7, 8 are not controlled by the IOC,
+    as they are under manual control.
+    """
+
+    def __init__(self, prefix: str, name: str = "") -> None:
+        with self.add_children_as_readables():
+            self.valve_states: DeviceVector[SignalR[ValveState]] = DeviceVector(
+                {i: epics_signal_r(ValveState, f"{prefix}V{i}:STA") for i in [1, 3]}
+            )
+            self.fast_valve_states: DeviceVector[SignalR[FastValveState]] = (
+                DeviceVector(
+                    {
+                        i: epics_signal_r(FastValveState, f"{prefix}V{i}:STA")
+                        for i in [5, 6]
+                    }
+                )
+            )
+
+        with self.add_children_as_readables(ConfigSignal):
+            self.valves_open: DeviceVector[SignalR[bool]] = DeviceVector(
+                {
+                    i: epics_signal_rw(bool, f"{prefix}V{i}:OPENSEQ")
+                    for i in [1, 3, 5, 6]
+                }
+            )
+
+            self.valve_control: DeviceVector[SignalR[ValveControlRequest]] = (
+                DeviceVector(
+                    {
+                        i: epics_signal_rw(ValveControlRequest, f"{prefix}V{i}:CON")
+                        for i in [1, 3, 5, 6]
+                    }
+                )
+            )
+
+        super().__init__(name)
+
+
+class Pump(StandardReadable):
+    def __init__(self, prefix: str, name: str = "") -> None:
+        with self.add_children_as_readables():
+            self.pump_position = epics_signal_r(float, prefix + "POS")
+            self.pump_forward_limit = epics_signal_r(
+                LimitSwitchState, prefix + "D74IN1"
+            )
+            self.pump_backward_limit = epics_signal_r(
+                LimitSwitchState, prefix + "D74IN0"
+            )
```

Comment:
Are the limits settable and required to be Read signals or are they read-once Config signals?

Reply:
how to check that?

Reply:
They're `epics_signal_r` read only from the ophyd layer, which means they're read-only from the ophyd layer. They could be derived from other signals, and therefore changing on the rate of every point of a scan, but I would infer from the name that they are hardware limits that don't change over the lifetime of the device, let alone within a scan. I would make them config signals.

Reply:
These signals are the state of limit switches for the pump so they will change during a scan if the pump is driven for a some length of time (approx 50s) in one direction. The name could be improved to `pump_..._limit_state`